### PR TITLE
add client#unsubscribe(subscriptions, [callback])

### PR DESCRIPTION
**Pull request by arden**
add client#unsubscribe(subscriptions, [callback])

### CODE REVIEW

File: lib/handlers/unsubscribe.js [hunk 1]
```diff
@@ -11,18 +11,24 @@ function UnsubscribeState (client, packet, finish, granted) {
...
+  //   broker.persistence.removeSubscriptions(client, packet.unsubscriptions, function (err) {
+  //     if (err) {
+  //       return done(err)
+  //     }
+  //
+  //     actualUnsubscribe(client, packet, done)
+  //   })
+  // } else {
+  //   actualUnsubscribe(client, packet, done)
+  // }
```
Comment (mcollina) [inline lib/handlers/unsubscribe.js]: What is this breaking? I fear this is because how we handle the close pipeline.

**Comment by mcollina** [comment]
Can you add unit test? Tests are failing as well.

File: lib/handlers/unsubscribe.js (hunk 1 above)
Comment (arden) [inline lib/handlers/unsubscribe.js]: i'm run it on my application is well. 
i will test again.

Comment (GavinDmello) [inline lib/handlers/unsubscribe.js]: Is the packet.messageId check important? It's not there in the changes

Comment (mcollina) [inline lib/handlers/unsubscribe.js]: I think the check is there to handle some weird situation. It was probably a hack I put in :/.

Comment (arden) [inline lib/handlers/unsubscribe.js]: why in subscribe.js

```
function handleSubscribe (client, packet, done) {
  var broker = client.broker
  var subs = packet.subscriptions
  var granted = []

  broker._series(
    new SubscribeState(client, packet, done, granted),
    doSubscribe,
    subs,
    completeSubscribe)
}
```
haven't check the pack.messageId?

Comment (GavinDmello) [inline lib/handlers/unsubscribe.js]:
```
https://github.com/mcollina/aedes/blob/master/lib/handlers/unsubscribe.js#L15
```
You're calling the persistence removeSubs every time now which wasn't happening previously

**Comment by mcollina** [comment]
Closing for no activity. Feel free to reopen in case you would like to keep working on it.

**Pull request closed without merging**